Sources/Provider/ItemProvider.swift

```diff
@@ -8,14 +8,14 @@
 
 import Foundation
 import Combine
-import Networking
+@preconcurrency import Networking
 import Persister
 
 /// Retrieves items from persistence or networking and stores them in persistence.
-public final class ItemProvider {
+public final class ItemProvider: Sendable {
```

Comment:
Should the type itself be marked as `@MainActor`? Or are there some parts that are not?

Reply:
I think no because a lot of the stuff doesn't need to be explicitly done from the main actor. Just the combine code calls more or less